src/dlgConnectionProfiles.cpp

```diff
@@ -1016,63 +1047,108 @@ void dlgConnectionProfiles::fillout_form()
-    if (onlyShownPredefinedProfiles.isEmpty()) {
-        const auto defaultGames = TGameDetails::keys();
-        for (auto& game : defaultGames) {
-            if (!deletedDefaultMuds.contains(game)) {
-                pItem = new QListWidgetItem();
-                auto details = TGameDetails::findGame(game);
-                setupMudProfile(pItem, game, (*details).description, (*details).icon);
+    const bool showAllPredefinedMuds = onlyShownPredefinedProfiles.isEmpty();
+    const auto defaultGames = TGameDetails::keys();
+    for (auto& game : defaultGames) {
+        auto details = TGameDetails::findGame(game);
+        auto pItem = new QListWidgetItem();
+        if (!game.compare(qsl("Mudlet self-test"), Qt::CaseInsensitive)) {
+            // We NEVER show this one - but we need it to be "in" the system
+            // so we can invoked it from the command-line with the "-p"
+            // option for tests
+            pItem->setData(csmHiddenRole, true);
+        } else {
+            if ((deletedDefaultMuds.contains(game))
+                ||(!showAllPredefinedMuds && !onlyShownPredefinedProfiles.contains(game, Qt::CaseInsensitive))) {
+                // We do not want this profile to show but we still need to
+                // keep track of it:
+                pItem->setData(csmHiddenRole, true);
+            } else {
+                pItem->setData(csmHiddenRole, false);
             }
+            setupMudProfile(pItem, game, (*details).description, (*details).icon);
+            reduceFontSize(pItem);
+            nameToItemMap.insert(pItem->data(csmNameRole).toString(), pItem);
         }
+    }
 
-#if defined(QT_DEBUG)
-        const QString mudServer = qsl("Mudlet self-test");
-        if (!deletedDefaultMuds.contains(mudServer) && !mProfileList.contains(mudServer)) {
-            mProfileList.append(mudServer);
-            pItem = new QListWidgetItem();
-            // Can't use setupMudProfile(...) here as we do not set the icon in the same way:
-            setItemName(pItem, mudServer);
-
+    // This badly named method actually creates icons for all the
+    // profiles found - and adds them to the nameToItemMap passed (by reference):
+    setProfileIcon(nameToItemMap);
+
+    // Now read the stored order - and the size we stored for this dialog:
+    QSize desiredDialogSize{};
+    QStringList existingProfilesOrderedList = loadProfilesOrderAndDlgSize(&desiredDialogSize);
+    // First make a pass through the QMap of QListWidgetItems - and identify
+    // those whose name is NOT in the stored, ordered, list of profiles, so we
+    // can place them first:
+    QMutableMapIterator<QString, QListWidgetItem*> itProfile(nameToItemMap);
+    while (itProfile.hasNext()) {
+        itProfile.next();
+        if (!existingProfilesOrderedList.contains(itProfile.key(), Qt::CaseInsensitive)) {
+            // This is a new one that we don't have an order position for, so
```

Comment:
This will happen when a new Predefined MUD gets added to Mudlet - and it'll stand out the first time it is shown!